File: Src2/Icm.Core/Tree/TraverseResult.vb
Namespace Icm.Tree

    Public Class TraverseResult(Of T)

        Public Sub New(result As T, level As Integer)
            Me.Result = result
            Me.Level = level
        End Sub

        Property Result As T

        Property Level As Integer

    End Class

    Public Class TraverseResult

        Public Shared Function Create(Of T)(result As T, level As Integer) As TraverseResult(Of T)
            Return New TraverseResult(Of T)(result, level)
        End Function

    End Class

End Namespace

File: Src/Icm.Core/Tree/TraverseResult.cs
namespace Icm.Tree
{

	public class TraverseResult<T>
	{

		public TraverseResult(T result, int level)
		{
			this.Result = result;
			this.Level = level;
		}

		public T Result { get; set; }

		public int Level { get; set; }

	}

	public class TraverseResult
	{

		public static TraverseResult<T> Create<T>(T result, int level)
		{
			return new TraverseResult<T>(result, level);
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
